extensions/ql-vscode/src/astViewer.ts

```diff
@@ -20,21 +20,23 @@ import { UrlValue, BqrsId } from './bqrs-cli-types';
 import { showLocation } from './interface-utils';
 import { isStringLoc, isWholeFileLoc, isLineColumnLoc } from './bqrs-utils';
 
+
 export interface AstItem {
   id: BqrsId;
   label?: string;
   location?: UrlValue;
   fileLocation?: Location;
-  parent: AstItem | RootAstItem;
-  children: AstItem[];
+  children: ChildAstItem[];
   order: number;
 }
 
-export type RootAstItem = Omit<AstItem, 'parent'>;
+export interface ChildAstItem extends AstItem {
+  parent: ChildAstItem | AstItem;
+}
 
-class AstViewerDataProvider implements TreeDataProvider<AstItem | RootAstItem> {
+class AstViewerDataProvider implements TreeDataProvider<ChildAstItem | AstItem> {
```

Comment:
Assuming this is covariant or invariant with respect to its type parameter, just `AstItem` would probably work, unless you're trying to enforce that everything may be a `ChildAstItem`.

Reply:
Sure. I'll fix that. Will need another 👀 .